res/controllers/Denon-MC7000-scripts.js

```diff
@@ -1036,26 +1216,85 @@ MC7000.HotCueLED = function(value, group) {
+    for (var deckIdx = 0; deckIdx < 4; deckIdx++) {
+        for (var samplerIdx = 1; samplerIdx <= 8; samplerIdx++) {
+            var sampNo = 0;
+            if (MC7000.PADModeSampler[deckIdx]) {
+                if (MC7000.SamplerQty === 16) {
+                    sampNo = (deckIdx % 2) * 8 + samplerIdx;
+                } else if (MC7000.SamplerQty === 32) {
+                    sampNo = deckIdx * 8 + samplerIdx;
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    var samplerIsNotPlaying = engine.getValue("[Sampler"+sampNo+"]", "play") === 0;
+                    if (samplerIsNotPlaying) {
+                        midi.sendShortMsg(0x94 + deckIdx, 0x14 + samplerIdx - 1, MC7000.padColor.samplerloaded);
+                        midi.sendShortMsg(0x94 + deckIdx, 0x1C + samplerIdx - 1, MC7000.padColor.samplerloaded); // shift pressed sets the same color
                     } else {
-                        midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.samplerplay);
+                        midi.sendShortMsg(0x94 + deckIdx, 0x14 + samplerIdx - 1, MC7000.padColor.samplerplay);
                     }
-                } else if (engine.getValue("[Sampler"+i+"]", "track_loaded") === 0) {
-                    midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.sampleroff);
+                } else if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 0) {
+                    midi.sendShortMsg(0x94 + deckIdx, 0x14 + samplerIdx - 1, MC7000.padColor.sampleroff);
+                    midi.sendShortMsg(0x94 + deckIdx, 0x1C + samplerIdx - 1, MC7000.padColor.sampleroff); // shift pressed sets the same color
                 }
             }
         }
     }
 };
 
+// Velocity Sampler LED
+MC7000.VelSampLED = function() {
+    for (var deckIdx = 0; deckIdx < 4; deckIdx++) {
+        for (var velSampIdx = 1; velSampIdx <= 8; velSampIdx++) {
+            var sampNo = 0;
+            if (MC7000.PADModeVelSamp[deckIdx]) {
+                if (MC7000.SamplerQty === 16) {
+                    sampNo = (deckIdx % 2) * 8 + velSampIdx; // use sampler 1-16 for deck 1 and deck 2 and sampler 1-16 for deck 3 and 4
+                } else if (MC7000.SamplerQty === 32) {
+                    sampNo = deckIdx * 8 + velSampIdx; // use sampler 1 - sampler 32 for deck 1 - 4
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    var samplerIsNotPlaying = engine.getValue("[Sampler"+sampNo+"]", "play") === 0;
+                    if (samplerIsNotPlaying) {
+                        midi.sendShortMsg(0x94 + deckIdx, 0x14 + velSampIdx - 1, MC7000.padColor.velsamploaded);
+                        midi.sendShortMsg(0x94 + deckIdx, 0x1C + velSampIdx - 1, MC7000.padColor.velsamploaded); // shift pressed sets the same color
```

Comment:
Please add comments what each of these two midi.sendShortMsg does. It's not obvious, why two are needed to be send here.

Reply:
The address of the pad changes when shift is pressed. To stop the play, shift and the pad button have to be pressed at the same time. From the timing, the execution of the function is behind the execution of the stop of the playing sampler. So it is possible that the shift button has already changed its state and the address switched back to 0x14. Otherwise still 0x1C. This addresses the above mentioned misbehave of the LED color.

Reply:
What do you think about a function `setPadColor` with arguments  for `deckIdx`, `offset`, `padIdx` and `colorCode` ?

Reply:
would be nice, but also affect the whole script. I will think about this.